Only in freetype-2.6/builds/amiga: makefile.os4
diff -r -u freetype-2.6/include/config/ftoption.h ttf/include/config/ftoption.h
--- freetype-2.6/include/config/ftoption.h	2015-04-22 03:57:28.000000000 -0500
+++ ttf/include/config/ftoption.h	2015-07-03 17:03:43.732494903 -0500
@@ -283,6 +283,48 @@
 /* #define FT_EXPORT_DEF(x)  x */
 
 
+/*
+  This hunk added for GraphicsMagick
+*/
+#ifndef  FT_EXPORT_DEF 
+
+/**
+ ** Borland C++ Builder defines
+ **/
+#  if defined(__BORLANDC__)
+#    if defined(_DLL)
+#      define _TTFDLL_
+#      define _TTFLIB_
+#    else
+#      undef _TTFDLL_
+#    endif   
+#  endif
+
+#  if defined(_MT) && defined(_DLL) && !defined(_TTFDLL_) && !defined(_LIB)
+#    define _TTFDLL_
+#  endif
+
+#  if defined(_TTFDLL_)
+#    if defined(_VISUALC_)
+#      pragma warning( disable : 4273 ) /* Disable the stupid dll linkage warnings */
+#      pragma warning( disable : 4018 )
+#      pragma warning( disable : 4244 )
+#      pragma warning( disable : 4142 )
+#    endif
+#    if !defined(_TTFLIB_)
+#      define FT_EXPORT_DEF( x ) __declspec(dllimport) x
+#      define FT_EXPORT( x ) __declspec(dllimport) x
+#    else
+#      define FT_EXPORT_DEF( x ) __declspec(dllexport) x
+#      define FT_EXPORT( x ) __declspec(dllexport) x
+#    endif
+#  else
+#    define FT_EXPORT_DEF( x ) extern x
+#    define FT_EXPORT( x ) extern x
+#  endif
+#endif
+
+
   /*************************************************************************/
   /*                                                                       */
   /* Glyph Postscript Names handling                                       */
Only in freetype-2.6/src/base: ftdebug.c
Only in ttf/src/base: ftdebug.c-renamed
